client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/GlobalOperationPrivilege.java

```diff
@@ -52,10 +53,16 @@
      *            The privilege definition.
      */
     public GlobalOperationPrivilege(String category, String operation, Map<String, Object> privilege) {
-        this.category = Objects.requireNonNull(category);
-        this.operation = Objects.requireNonNull(operation);
+        if (Strings.hasText(category) == false) {
+            throw new IllegalArgumentException("category is required");
+        }
+        this.category = category;
+        if (Strings.hasText(operation) == false) {
+            throw new IllegalArgumentException("operation is required");
+        }
```

Comment:
I specifically wanted only `null` checks.
On the server, these values have length and format checks (loosely speaking), and I have purposefully refrained from any value check on client side, except null checks. It's a purist view, I know, but I don't see value in changing this behavior.

Reply:
Are you especially bothered by these?

Reply:
🙏 thank you for tolerating me.